### PR TITLE
Tracker updates for version 1.23x

### PR DESCRIPTION
**Pull request by mauvealerts**
As far as I know, there are two things preventing this from being releasable
* Updating entity type IDs
* More thoroughly testing entity lookup

For entity lookup testing, I currently have it both do the lookup the hash way, and also a full table scan. Mismatches are logged. My other ideas are:
* Check if we actually probe multiple entries in the hash version
* Do a full scan and do the fast lookup for every entry

### CODE REVIEW

**Comment by mauvealerts** [comment]
I added code to periodically scan the entire entity map and lookup every uid in there. It didn't find any mismatches after sitting in camp for 30 minutes and having entity uids over 40,000. I've removed all the code that did full scans.

I think this is in a releasable state. I'd kinda like  to add unit tests, and maybe simplify TP code using new `abs_x` and `abs_y`, but those seem like "nice to have"

**Review by gmjosack — APPROVED** [review]
left one comment but otherwise 🤞

File: src/modlunky2/mem/state.py [hunk 1]
```diff
@@ -170,14 +169,12 @@ class State:
...
-        unordered_map(sc_uint32, poly_pointer(dc_struct)),
-        default_factory=DictUnorderedMap,
+    run_recap_flags: RunRecapFlags = struct_field(0xA34, sc_uint32, default=0)
+    hud_flags: HudFlags = struct_field(0xA50, sc_uint32, default=0)
+    time_level: int = struct_field(0xA44, sc_uint32, default=0)
+    presence_flags: PresenceFlags = struct_field(0xA54, sc_uint32, default=0)
+    next_entity_uid: int = struct_field(0x12E0, sc_uint32, default=0)
+    items: Optional[Items] = struct_field(0x12F0, pointer(dc_struct), default=None)
+    instance_id_to_pointer: UidEntityMap = struct_field(
+        0x1348, uid_entity_map, default_factory=DictUnorderedMap
```
Comment (gmjosack) [inline src/modlunky2/mem/state.py:179]: should this still be DictUnorderedMap for the default factory?

Comment (mauvealerts) [reply src/modlunky2/mem/state.py]: Yeah, it's a bit more general. I renamed this to `DictMap` and moved it from `msvc` to `model` to better reflect that

**Pull request merged**